packages/app/src/systems/Asset/components/AssetItem/AssetItem.tsx

```diff
@@ -1,48 +1,141 @@
 import { cssObj } from '@fuel-ui/css';
-import { Avatar, CardList, Flex, Heading, Text, Tooltip } from '@fuel-ui/react';
-import type { Coin } from '@fuel-wallet/types';
+import {
+  Avatar,
+  Button,
+  CardList,
+  Copyable,
+  Flex,
+  Heading,
+  Icon,
+  IconButton,
+  Text,
+  Tooltip,
+} from '@fuel-ui/react';
+import type { AssetAmount } from '@fuel-wallet/types';
 import { bn } from 'fuels';
 import type { FC } from 'react';
+import { useNavigate } from 'react-router-dom';
 
-import { getAssetInfoById } from '../../utils';
+import { AssetRemoveDialog } from '../AssetRemoveDialog';
 
 import { AssetItemLoader } from './AssetItemLoader';
 
-import { AmountVisibility } from '~/systems/Core';
+import { AmountVisibility, Pages, shortAddress } from '~/systems/Core';
 import { useBalanceVisibility } from '~/systems/Core/hooks/useVisibility';
 
 export type AssetItemProps = {
-  asset: Coin;
+  asset: AssetAmount;
+  isHidden?: boolean;
+  showActions?: boolean;
+  onRemove?: (assetId: string) => void;
+  onEdit?: (assetId: string) => void;
 };
 
 type AssetItemComponent = FC<AssetItemProps> & {
   Loader: typeof AssetItemLoader;
 };
 
-export const AssetItem: AssetItemComponent = ({ asset }) => {
-  const { symbol, name, imageUrl } = getAssetInfoById(asset.assetId, asset);
+export const AssetItem: AssetItemComponent = ({
+  asset,
+  isHidden,
+  showActions,
+  onRemove,
+  onEdit,
+}) => {
+  const navigate = useNavigate();
   const { visibility } = useBalanceVisibility();
 
-  const rightEl = (
-    <Tooltip content={bn(asset.amount).format()}>
-      <Text css={{ fontSize: '$sm', fontWeight: '$semibold' }}>
-        <AmountVisibility value={asset.amount} visibility={visibility} />{' '}
-        {symbol}
-      </Text>
-    </Tooltip>
-  );
+  if (isHidden) return null;
+  const {
+    symbol,
+    name = '',
+    imageUrl,
+    amount,
+    assetId,
+    isCustom,
+  } = asset || {};
```

Comment:
We should move the isHidden to an upper component and filter the hidden assets to avoid unnecessary render.

Reply:
in `AccountItem` it's done in the exact same way, if you want we can open an issue to fix it there later...

I will fix only in `AssetItem` for now